overlord/snapstate/snapstate_test.go

```diff
@@ -176,9 +185,11 @@ func verifyInstallUpdateTasks(c *C, opts, discards int, ts *state.TaskSet, st *s
 	}
 	expected = append(expected,
 		"set-auto-aliases",
-		"setup-aliases",
-		"start-snap-services",
-	)
+		"setup-aliases")
+	if opts&install != 0 {
+		expected = append(expected, "run-hook")
```

Comment:
It's not enough to test it like that, because we don't know which hook is being run now that we can have many of them. We need to make sure that we have a proper install hooks on install, a proper remove hook on remove, and we also need to make sure that these are only added when first installing and last-removing the snap, and not otherwise. These checks should be done on the tests that verify that particular aspect (installation and removal), without adding custom logic to other tests like was done for the SkipConfigure one below.